### ISSUE TITLE
Date picker missing disabled styles in all variants

### ISSUE THREAD

**Issue by emyarod** [carbon-design-system/carbon#2498]
below are screenshots of the disabled state for each date picker variant. the placeholder text color and icon colors are incorrect. This likely needs changes in the vanilla repo

simple

![image](https://user-images.githubusercontent.com/8265238/55572243-d1854500-56cc-11e9-8d1b-a136b9ebd067.png)

single with calendar

![image](https://user-images.githubusercontent.com/8265238/55572279-de099d80-56cc-11e9-96c3-724a0eb4c68d.png)

range with calendar

![image](https://user-images.githubusercontent.com/8265238/55572308-ea8df600-56cc-11e9-8c93-0ad3a378e3da.png)

range with calendar and min/max dates

![image](https://user-images.githubusercontent.com/8265238/55572319-f37ec780-56cc-11e9-9e16-fe8f73ab193a.png)

**Comment by laurenmrice** [comment]
Vanilla looks correct but React has some issues regarding the placeholder text and disabled states. The icons in the disabled state should also be `disabled-02` right now it is `icon-01`.

Here is a spec of the token colors that should be used for the following states. 
![Artboard](https://user-images.githubusercontent.com/43969356/62228128-b5e03180-b382-11e9-92ab-73daba5f2c93.png)
